android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js

```diff
@@ -5,16 +5,17 @@
 // This background script is needed to update the current tab
 // and activate reader view.
 
-browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
+browser.runtime.onMessage.addListener(message => {
   switch (message.action) {
      case 'addSerializedDoc':
         browser.storage.session.set({ [message.id]: message.doc });
-        break;
+        return Promise.resolve();
      case 'getSerializedDoc':
-       let doc = await browser.storage.session.get(message.id);
-       browser.storage.session.remove(message.id);
-       sendResponse(doc);
```

Comment:
@boek I wanted to mention this explicitly at least here: when a runtime.onMessage listener is an async function then when it is called its return value is always going to a promise and runtime.onMessage will in that case use the value resolved by the returned promise as the value to return to the sender of the message being handled (and sendResponse is ignored, see [here](https://searchfox.org/mozilla-central/rev/3fe52b82b0a00829d0a8224b4bd767cb20b7a27d/toolkit/components/extensions/ExtensionChild.sys.mjs#210-216)).